README.md

```diff
@@ -2,36 +2,121 @@
 
 [![Build Status](https://travis-ci.org/namshi/reconfig.svg?branch=travis)](https://travis-ci.org/namshi/reconfig)
 
-Handy and _strongly_ opinionated config helper on top of [Reconfig](https://github.com/namshi/reconfig).
+Handy and _strongly_ opinionated config helper on top of [reconfig](https://github.com/namshi/reconfig).
 
-## Installation
+## So what about it?
+Here in Namshi we like our configuration incremental, over-ridable, and defined in a compact way. As well as we like to don't go nuts when we clone a project repo for the 1st time.
+This is way we came up with tools like [reconfig](https://github.com/namshi/reconfig) and [file-ensure](https://github.com/namshi/node-file-ensure). With nmConfig we put all this things together in a convenient lib that turns all of this in a matter of a `require()` instruction.
 
-Install this library via [NPM](https://www.npmjs.org/package/node-nmconfig):
+## What does it do:
 
-``` bash
-npm install node-nmconfig
+- Automatically creates a `config/` directory and the needed files if are not present:
+
+```
+config/
+     |- base.yml
+     |- dev.example.yml
+     |- staging.yml
+     |- live.yml
 ```
+- Compiles your config files, figure out your environment, and applies the needed merging on the `base.yml` one:
+
+```yml
+
+#base.yml
+characters:
+  yoda: jedi
+  anakin: jedi
+  obiWan: jedi
+
+#dev.yml
+characters:
+  anakin: sith
 
-## Usage
-```javascript
-var config = require('node-nmconfig');
 ```
 
-If you need it on the client side we highly recommend
-[browserify](http://browserify.org/).
+will result in:
 
-## Tests
+```javascript
+{
+  characters: {
+    yoda: 'jedi',
+    anakin: 'sith',
+    obiWan: 'jedi'
+  }
+}
+```
 
-This library is tested through mocha, simply run either
+- returns you a `reconfig` instance:
+```javascript
+console.log(config.get('characters.anakin'));
 
+// ==> 'sith'
 ```
-mocha
+
+- Figures out a `recofig`'s env overrider prefix from your `package.json`:
```

Comment:
nope, that's another thing.
It's reconfig thing: overlays coming form the env overriding config values